proposals/lip-define_new_transaction_schema.md

```diff
@@ -0,0 +1,258 @@
+
+### JSON Schema
+
+Transactions are serialized using `transactionSchema` given below.
+
+
+```java
+transactionSchema = {
+    "type": "object",
+    "required": [
+        "module",
+        "command",
+        "nonce",
+        "fee",
+        "senderPublicKey",
+        "params",
+        "signatures"
+    ],
+    "properties": {
+        "module": {
+            "dataType": "string",
+            "minLength": MIN_MODULE_NAME_LENGTH, 
+            "maxLength": MAX_MODULE_NAME_LENGTH,
+            "fieldNumber": 1
+        },
+        "command": {
+            "dataType": "string",
+            "minLength": MIN_COMMAND_NAME_LENGTH, 
+            "maxLength": MAX_COMMAND_NAME_LENGTH,
+            "fieldNumber": 2
+        },
+        "nonce": {
+            "dataType": "uint64",
+            "fieldNumber": 3
+        },
+        "fee": {
+            "dataType": "uint64",
+            "fieldNumber": 4
+        },
+        "senderPublicKey": {
+            "dataType": "bytes",
+            "length": ED25519_PUBLIC_KEY_LENGTH,
+            "fieldNumber": 5
+        },
+        "params": {
+            "dataType": "bytes",
+            "fieldNumber": 6
+        },
+        "signatures": {
+            "dataType": "array",
+            "items": {
+                "dataType": "bytes",
+                "length": ED25519_SIGNATURE_LENGTH 
+            },
+            "fieldNumber": 7
+        }
+    }
+}
+```
+
+
+#### Validation
+
+For a transaction `trs` to be valid, it must satisfy the following:
+
+* `trs` must follow the `transactionSchema`. 
+* `trs.params` is of length less than or equal to `MAX_PARAMS_SIZE` .
+
+The validity of each transaction is verified in the [static validation stage][lip-0055#block-processing] of the block processing. 
+
+
+### Serialization
+
+The serialization of an object of type `Transaction` is described in the following pseudocode.
+
+```python
+def encode(paramsSchema: LiskJSONSchema, trs: Transaction) -> bytes:
+    trs.params = encode(paramsSchema,trs.params)
+    return encode(transactionSchema,trs)
+```
+
+### Deserialization
+
+Consider a binary message `trsMsg`, corresponding to a serialized transaction. The deserialization procedure is as follows:
+
+```python
+def decode(paramsSchema: LiskJSONSchema, trsMsg: bytes) -> Transaction: 
+    trsData = decode(transactionSchema,trsMsg)
+    trsData.params = decode(paramsSchema,trsData.params)
+    return trsData
+```
+
+### Transaction Signature Calculation
+
+Consider a data structure `unsignedTrsData` representing a valid `Transaction` object in which the signatures array is initialized to the default value (an empty array). The following function calculates a signature of the object on a certain chain with secret key `sk`. 
+
+```python
+def computeTransactionSignature(sk: PrivateKeyEd25519, unsignedTrsData: Transaction, networkIdentifier: bytes) -> SignatureEd25519:
```

Comment:
Alternatively, we could just pass the transaction object and set the signatures to empty explicitly before signing it

Reply:
I think I prefer that approach as it is more robust.

Reply:
I updated the function as suggested.